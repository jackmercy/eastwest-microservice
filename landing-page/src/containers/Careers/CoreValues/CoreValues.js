import { memo, useMemo } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { useMediaQuery } from 'react-responsive';

import BREAKPOINTS from '../../../constants/BREAKPOINTS';

import WebCoreValues from './WebCoreValues';
import MobileCoreValues from './MobileCoreValues';
import useStyles from './coreValues.styles';

const CoreValues = ({ id }) => {
  const intl = useIntl();
  const classes = useStyles();
  const isMobile = useMediaQuery({
    query: `(max-width: ${BREAKPOINTS.maxMd}px)`,
  });
  const coreValues = useMemo(
    () => [
      {
        key: 'strive-for-excellence',
        title: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.01.TITLE',
        }),
        titleTranslationId: 'CAREERS.CORE_VALUE.01.TITLE',
        description: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.01.TEXT',
        }),
        image: '/static/images/careers/core-value-01.png',
        srcSet: [
          '/static/images/careers/core-value-01-mobile.png 500w',
          '/static/images/careers/core-value-01.png 2000w',
        ].join(', '),
      },
      {
        key: 'think-in-solutions',
        title: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.02.TITLE',
        }),
        titleTranslationId: 'CAREERS.CORE_VALUE.02.TITLE',
        description: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.02.TEXT',
        }),
        image: '/static/images/careers/core-value-02.png',
        srcSet: [
          '/static/images/careers/core-value-02-mobile.png 500w',
          '/static/images/careers/core-value-02.png 2000w',
        ].join(', '),
      },
      {
        key: 'be-inclusive',
        title: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.03.TITLE',
        }),
        titleTranslationId: 'CAREERS.CORE_VALUE.03.TITLE',
        description: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.03.TEXT',
        }),
        image: '/static/images/careers/core-value-03.png',
        srcSet: [
          '/static/images/careers/core-value-03-mobile.png 500w',
          '/static/images/careers/core-value-03.png 2000w',
        ].join(', '),
      },
      {
        key: 'work-towards-productivity',
        title: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.04.TITLE',
        }),
        titleTranslationId: 'CAREERS.CORE_VALUE.04.TITLE',
        description: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.04.TEXT',
        }),
        image: '/static/images/careers/core-value-04.png',
        srcSet: [
          '/static/images/careers/core-value-04-mobile.png 500w',
          '/static/images/careers/core-value-04.png 2000w',
        ].join(', '),
      },
    ],
    [intl],
  );

  return (
    <div id={id} className={classes.root}>
      {!isMobile ? (
        <WebCoreValues data={coreValues} />
      ) : (
        <MobileCoreValues data={coreValues} />
      )}
    </div>
  );
};
CoreValues.propTypes = { id: PropTypes.string.isRequired };

export default memo(CoreValues);
