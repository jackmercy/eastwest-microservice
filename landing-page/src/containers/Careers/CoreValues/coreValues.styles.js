import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: { marginTop: rem(160) },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    root: { marginTop: rem(64) },
  },
});
