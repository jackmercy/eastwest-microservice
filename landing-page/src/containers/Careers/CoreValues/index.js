import dynamic from 'next/dynamic';

export default dynamic(() => import('./CoreValues'), { ssr: false });
