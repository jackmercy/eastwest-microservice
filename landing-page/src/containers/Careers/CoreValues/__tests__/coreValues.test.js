import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CoreValues from '../CoreValues';

describe('Testing Careers > CoreValues', () => {
  const props = { id: 'our-mission' };

  it('should render match snapshot', () => {
    const wrapper = shallow(<CoreValues {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
