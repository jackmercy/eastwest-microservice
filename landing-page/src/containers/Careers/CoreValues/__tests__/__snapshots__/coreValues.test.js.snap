// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Careers > CoreValues should render match snapshot 1`] = `
<div
  className="root-0-2-1"
  id="our-mission"
>
  <Memo(WebCoreValues)
    data={
      Array [
        Object {
          "description": "Given our responsibility to each other and our customers, behaving ethically is a critical, and elemental, part of our success. Our clients are pouring their heart and soul into their product. There is no alternative for us but do the same.",
          "image": "/static/images/careers/core-value-01.png",
          "key": "strive-for-excellence",
          "srcSet": "/static/images/careers/core-value-01-mobile.png 500w, /static/images/careers/core-value-01.png 2000w",
          "title": "Strive for excellence",
          "titleTranslationId": "CAREERS.CORE_VALUE.01.TITLE",
        },
        Object {
          "description": "We start from a baseline of common sense, feedback loops and retrospectives that show us how to improve. Problems are meant to be challenged. Sharing that knowledge with each other, and acting on it, is essential for our Talents.",
          "image": "/static/images/careers/core-value-02.png",
          "key": "think-in-solutions",
          "srcSet": "/static/images/careers/core-value-02-mobile.png 500w, /static/images/careers/core-value-02.png 2000w",
          "title": "Think in solutions",
          "titleTranslationId": "CAREERS.CORE_VALUE.02.TITLE",
        },
        Object {
          "description": "In a competitive landscape full of talented people, empathy and kindness go a long way. Our Talents have built a culture where inclusion and respect are a defining part of our identity.",
          "image": "/static/images/careers/core-value-03.png",
          "key": "be-inclusive",
          "srcSet": "/static/images/careers/core-value-03-mobile.png 500w, /static/images/careers/core-value-03.png 2000w",
          "title": "Be inclusive",
          "titleTranslationId": "CAREERS.CORE_VALUE.03.TITLE",
        },
        Object {
          "description": "We hire intelligent people to tell us what to do, not to tell them what to do! You and your team mates are part of the solution. Work together and continuously strive to be productive.",
          "image": "/static/images/careers/core-value-04.png",
          "key": "work-towards-productivity",
          "srcSet": "/static/images/careers/core-value-04-mobile.png 500w, /static/images/careers/core-value-04.png 2000w",
          "title": "Work towards productivity",
          "titleTranslationId": "CAREERS.CORE_VALUE.04.TITLE",
        },
      ]
    }
  />
</div>
`;
